SourceX/main.cpp

```diff
@@ -24,6 +28,26 @@ static std::string build_cmdline(int argc, char **argv)
 
 int main(int argc, char **argv)
 {
+#ifdef PLATFORM_CTR
+	bool isN3DS;
+
+	APT_CheckNew3DS(&isN3DS);
+	if(isN3DS)
+		osSetSpeedupEnable(true);
+	
+	romfsInit();
+	atexit(romfsExit());
+	
+	if (SDL_Init(SDL_INIT_TIMER | SDL_INIT_VIDEO ) != 0) {
+		printf("Failed to start SDL!\n");
+		return -1;
+	}
+	SDL_Surface *screen;
+	screen = SDL_SetVideoMode(400, 240, 0, SDL_SWSURFACE | SDL_CONSOLEBOTTOM); //Init console

```

Comment:
Is an explicit resolution necessary?

Custom flags can be passed via CMake, `-DSDL1_VIDEO_MODE_FLAGS='SDL_SWSURFACE|SDL_CONSOLEBOTTOM'`